core/src/main/java/hudson/security/HudsonPrivateSecurityRealm.java

```diff
@@ -279,13 +281,24 @@ public void doCreateAccountByAdmin(StaplerRequest req, StaplerResponse rsp) thro
      * This can be run by anyone, but only to create the very first user account.
      */
     public void doCreateFirstAccount(StaplerRequest req, StaplerResponse rsp) throws IOException, ServletException {
-        if(hasSomeUser()) {
+        boolean inSetup = !Jenkins.getInstance().getInstallState().isSetupComplete();
+        if(!inSetup && hasSomeUser()) {
             rsp.sendError(SC_UNAUTHORIZED,"First user was already created");
             return;
         }
-        User u = createAccount(req, rsp, false, "firstUser.jelly");
+        String view = "firstUser.jelly";
+        User admin = null;
+        if(inSetup) {
+            admin = getUser(SetupWizard.initialSetupAdminUserName);
+            view = "setupWizardFirstUser.jelly";
+        }
+        User u = createAccount(req, rsp, false, view);
         if (u!=null) {
             tryToMakeAdmin(u);
+            if(admin != null) {
+                admin.delete();
```

Comment:
@jglick I think the intention is to have a temporary admin account for the setup only, and then have a complete user registration (with email, full name, and custom password) afterwards.